### PR TITLE
Golang: Make VaaS client closeable

### PR DESCRIPTION
**Pull request by rageagainsthepc**
Right now it is not possible to cancel the socket initialization because the context that is passed via `Connect()` is also used for closing the socket. It is more idiomatic in golang to let the `VaaS` client implement the `io.Closer` interface and use that to close the connection explicitly.

### CODE REVIEW

**Review by unglaublicherdude — APPROVED** [review]

File: golang/vaas/go.mod
Comment (unglaublicherdude) [inline golang/vaas/go.mod:3]: Please let that stay on 1.21 as this was requested by the outlook team. the problem is, that in the workflow the go-setup does not use the latest version of 1.21 I will have a look at this.

File: golang/vaas/pkg/vaas/vaas.go
Comment (cavus700) [inline golang/vaas/pkg/vaas/vaas.go:649]: I would prefer to move the go routine inside `readWebSocket` for two reasons:
1. If you do it like this you always have to start the function in a go routing, otherwise the call blocks and you can not read from the channel
2. Not sure if this is an "offical idiom" but for me the one who creates the channel should always be the one who closes it, too. When you pass a channel to a function and the function closes the channel it is some unexpected behaviour and can cause crashes.

```suggestion
	errorChan := v.readWebSocket()
```

**Comment by cavus700** [comment]
Keep in mind that this is a breaking change and will not crash after a simple update in other projects. Furthermore the golang version of vaas does not have correct golang versioning it is unlikely any one will notice it during an update. I don't know how many stakeholders this project has and how you deal with breaking changes but I would not merge this even if the code looks good to me.

**Comment by unglaublicherdude** [comment]
@cavus700 the [Versioning was fixed](https://github.com/GDATASoftwareAG/vaas/pull/491).

**Comment by cavus700** [comment]
Nice, I didn't know about the versioning but in this case this changes would require a v2 for me.

File: golang/vaas/pkg/vaas/vaas.go
Comment (rageagainsthepc) [reply golang/vaas/pkg/vaas/vaas.go:649]: regarding `2.`: So you are basically saying go routins should be anonymous?

Comment (rageagainsthepc) [reply golang/vaas/pkg/vaas/vaas.go:649]: I tried to alter as little as possible when it comes to design choices. I would leave this up to @unglaublicherdude if that's ok. Feel free to alter this PR if you think this is a sensible suggestion.

Comment (unglaublicherdude) [reply golang/vaas/pkg/vaas/vaas.go:649]: I don't have a heavy opinion here, because we don't really use this SDK in bigger projects. So while you are the ones actually using the SDK I will let you make the decision. But that's just me, maybe @GermanCoding @pstadermann @doxthree or @lennartdohmann have a different opinion on that.

Comment (rageagainsthepc) [reply golang/vaas/pkg/vaas/vaas.go]: I just went ahead and implemented the requested changes in order to avoid further discussion and get this merged quickly. ;)

Comment (unglaublicherdude) [reply golang/vaas/pkg/vaas/vaas.go]: @cavus700 @rageagainsthepc do you agree on this PR now? If so, I don't mind merging it.

Comment (cavus700) [reply golang/vaas/pkg/vaas/vaas.go]: The code itself looks good to me but in my opinion these changes still require a v2 and I would not merge it without. 
Since this is not my project and I don't know all stakeholders the decision is up to you.

**Comment by unglaublicherdude** [comment]
And again I forgot that you have to put the v2 stuff into its own folder.
@rageagainsthepc will you do that or should I? Its go specific, but on Major releases it requires to have the new major release in its own folder. 

So the changes you made have to be in the folder golang/vaas/v2 wich should also be an own module.

An example is the [go-gax](https://github.com/googleapis/gax-go) repository.

**Pull request merged**
